syne_tune/optimizer/schedulers/searchers/searcher.py

```diff
@@ -577,11 +598,53 @@ def get_batch_configs(self, batch_size: int, **kwargs):
                 configs.append(config)
         return configs
 
-    def _validate_config_space(self, config_space: dict):
-        # GridSearcher only supports Categorical hyperparameters for now
-        for hp_range in config_space.values():
-            if isinstance(hp_range, Domain):
-                assert isinstance(hp_range, Categorical)
+    def _validate_config_space(self, config_space: dict, num_samples: dict):
+        """
+        Validates config_space from two aspects: first, that all hyperparameters
+        are of acceptable types (i.e. float, integer, categorical). Second, num_samples is specified
+        for float hyperparameters. Num_samples for categorical variables are ignored as all of their
+        values is used in GridSearch. Num_samples for integer variables is optional, if specified it
+        will be used and will be capped at their range length.
+        Args:
+            config_space: The configuration space that defines search space grid.
+            num_samples: Number of samples for each hyperparmaters.Only required for float hyperparameters.
+        """
+        if num_samples is None:
+            num_samples = {}
+        self.num_samples = num_samples
+        for hp, hp_range in config_space.items():
+            if isinstance(hp_range, Float):
+                if hp not in self.num_samples:
+                    self.num_samples[hp] = GridSearcher.DEFAULT_NSAMPLE
+                    logger.warning(
+                        "number of samples is required for {}. By default, {} is set as number of samples".format(
+                            hp, GridSearcher.DEFAULT_NSAMPLE
+                        )
+                    )
+            # n_sample for integer hp must be capped at length of the range
+            if isinstance(hp_range, Integer):
+                if hp in self.num_samples:
+                    if self.num_samples[hp] > len(hp_range):
+                        self.num_samples[hp] = GridSearcher.DEFAULT_NSAMPLE
+                        logger.info(
+                            'number of samples for "{}" is larger than its range. By default, {} is set as number of samples'.format(
+                                hp, GridSearcher.DEFAULT_NSAMPLE
+                            )
+                        )
+                else:
+                    self.num_samples[hp] = GridSearcher.DEFAULT_NSAMPLE
+                    logger.info(
```

Comment:
I'd remove this warning. If the user does not say anything, they don;'t want to see warnings.

Reply:
good point